backend/app/tests/memory_benchmark.py

```
@@ -0,0 +1,172 @@
+
+    # Force import of the heavy dependency modules
+    from fastapi import Depends, FastAPI
+    from fastapi.dependencies.models import Dependant
+
+    app = FastAPI()
+
+    # Register endpoints with various dependency patterns
+    for i in range(50):
+        # Create endpoints with nested deps to expand the dependency graph
+        async def dep_a() -> dict[str, int]:
+            return {"a": i}
```

Ruff (0.15.21)

[warning] Function definition does not bind loop variable `i`

(B023)